pkg/controller/releasemanager/scaling/geometric_test.go

```diff
@@ -0,0 +1,101 @@
+package scaling
+
+import (
+	picchu "go.medium.engineering/picchu/pkg/apis/picchu/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	testify "github.com/stretchr/testify/assert"
+	"go.medium.engineering/picchu/pkg/controller/releasemanager/scaling/mocks"
+)
+
+func prepareGeometricMock(ctrl *gomock.Controller, isReconciled bool, currentPercent, peakPercent, factor, max, start int, delay time.Duration, lastUpdated time.Time) ScalableTarget {
+	m := mocks.NewMockScalableTarget(ctrl)
+	releaseInfo := picchu.ReleaseInfo{
+		Eligible:        false,
+		Max:             uint32(max),
+		ScalingStrategy: picchu.ScalingStrategyGeometric,
+		GeometricScaling: picchu.GeometricScaling{
+			Start:  uint32(start),
+			Factor: uint32(factor),
+			Delay:  &metav1.Duration{Duration: delay},
+		},
+		LinearScaling: picchu.LinearScaling{},
+		Rate:          picchu.RateInfo{},
+		Schedule:      "",
+		TTL:           0,
+	}
+
+	m.
+		EXPECT().
+		CurrentPercent().
+		Return(uint32(currentPercent)).
+		AnyTimes()
+	m.
+		EXPECT().
+		IsReconciled(gomock.Any()).
+		Return(isReconciled).
+		AnyTimes()
+	m.
+		EXPECT().
+		PeakPercent().
+		Return(uint32(peakPercent)).
+		AnyTimes()
+	m.
+		EXPECT().
+		ReleaseInfo().
+		Return(releaseInfo).
+		AnyTimes()
+	m.
+		EXPECT().
+		LastUpdated().
+		Return(lastUpdated).
+		AnyTimes()
+
+	return m
+}
+
+func TestGeometricScaling(t *testing.T) {
+	assert := testify.New(t)
+	ctrl := gomock.NewController(t)
+
+	defer ctrl.Finish()
+
```

Comment:
have you seen the table test pattern?

```go
cases := []struct{
	CurrentPercent int
	IsReconciled bool
	// ...
}{
	{
		CurrentPercent: 0,
		IsReconciled: false,
		// ...
	},
}

for _, c := range cases {
	m := mocks.NewMockScalableTarget(ctrl)
	// ...
}
```

I would have a lot easier time understanding these tests if the cases were defined like that, using structs instead of 9 unnamed function arguments.